packages/manager/src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/AddNodePoolDrawer.tsx

```diff
@@ -109,39 +110,28 @@ export const AddNodePoolDrawer = (props: Props) => {
     }
   }, [error]);
 
+  React.useEffect(() => {
+    if (selectedTypeInfo?.count === 0) {
+      setSelectedTypeInfo(undefined);
+    }
+  }, [selectedTypeInfo]);
+
   const resetDrawer = () => {
-    const newTypes = _types.map((thisType) => {
-      return {
-        ...thisType,
-        count: 0,
-      };
-    });
-    setNewType(newTypes);
-    setSelectedType(undefined);
+    setSelectedTypeInfo(undefined);
   };
 
   const updatePlanCount = (planId: string, newCount: number) => {
-    const newTypes = _types.map((thisType: any) => {
-      if (thisType.id === planId) {
-        return { ...thisType, count: newCount };
-      }
-      return { ...thisType, count: 0 };
-    });
-    setNewType(newTypes);
-    // If everything's empty, we need to reset the selected type.
-    if (newTypes.every((thisType) => thisType.count === 0)) {
-      setSelectedType(undefined);
-    } else {
-      setSelectedType(planId);
-    }
+    setSelectedTypeInfo({ planId, count: newCount });
```

Comment:
Would doing this allow us to get rid of the `useEffect` that runs `setSelectedTypeInfo(undefined)`

```suggestion
    if (newCount === 0) {
      setSelectedTypeInfo(undefined);
    } else {
      setSelectedTypeInfo({ planId, count: newCount });
    }
```